db/schema.rb

```diff
@@ -223,6 +223,8 @@
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
     t.string "visit_token"
+    t.text "properties"
```

Comment:
Another option might also be to add `, default: "[]"` here to prevent old reports from failing without the multiple checks. I'm totally fine with your method as well though.

Reply:
I think I'll go ahead and add that to keep the helper methods cleaner.